### PR TITLE
DM-35347: Switch to pyproject.toml

### PR DESCRIPTION
**Pull request by mwittgen**
## Checklist

- [ ] ran Jenkins
- [ ] added a release note for user-visible changes to `doc/changes`

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,7 +1,64 @@
...
+    "lsst-resources [s3]",
+    "lsst-utils",
+    "lsst-daf-butler",
+    "lsst-pex-config",
+    "pydantic",
+    "networkx",
+    "pyyaml >= 5.1",
+    "numpy >= 1.17",
+]
+
```
Comment (timj) [inline pyproject.toml:34]: remove line

File: pyproject.toml [hunk 2]
```diff
@@ -1,7 +1,64 @@
...
+[project.optional-dependencies]
+test = [
+    "pytest >= 3.2",
+    "flake8 >= 3.7.5",
+    "pytest-flake8 >= 1.0.4",
+    "pytest-openfiles >= 0.5.0",
+]
+
+
```
Comment (timj) [inline pyproject.toml:48]: remove line

**Review by timj — APPROVED** [review]
Looks fine. I think the mypy fixes are waiting on #259

**Comment by mwittgen** [comment]
Added two commits with the `black` and `isort` fixes, running jenkins right now.

**Pull request merged**